proxy/http/HttpSM.cc

```diff
@@ -6560,6 +6559,12 @@ HttpSM::kill_this()
   reentrancy_count--;
   ink_release_assert(reentrancy_count == 0);
 
+  // Delay the close of the user agent session, so the close session
+  // occurs after the close transaction
+  if (ua_session != NULL)
```

Comment:
Not what I see for non-VC errors (in fact, there really is no reason to close the connection on non-VC errors).

For example, https://github.com/apache/trafficserver/blob/master/proxy/http/HttpSM.cc#L1424